src/azure-cli/azure/cli/command_modules/servicefabric/_validators.py

```diff
@@ -107,6 +109,199 @@ def validate_create_managed_cluster(cmd, namespace):
+    client = servicefabric_managed_client_factory(cmd.cli_ctx)
+    service = _safe_get_resource(client.services.get,
+                                 (namespace.resource_group_name, namespace.cluster_name,
+                                  namespace.application_name, namespace.service_name))
+    if service.properties.service_kind.lower() == ServiceKind.STATELESS.lower():
+        if namespace.target_replica_set_size or namespace.min_replica_set_size:
+            raise CLIError("--target-replica-set-size and --min-replica-set-size should only be use with "
+                           "--state stateful")
+        if namespace.instance_count is not None:
+            namespace.instance_count = int(namespace.instance_count)
+    elif service.properties.service_kind.lower() == ServiceKind.STATEFUL.lower():
+        if namespace.instance_count:
+            raise CLIError("Unexpected parameter --instance-count should only be use with --state stateless")
+        if namespace.target_replica_set_size is not None:
+            namespace.target_replica_set_size = int(namespace.target_replica_set_size)
+        if namespace.min_replica_set_size is not None:
+            namespace.min_replica_set_size = int(namespace.min_replica_set_size)
+    else:
+        raise CLIError("Invalid --state '%s': service state is not valid." % service.properties.service_kind)
+
+
+def validate_create_managed_service_load_metric(cmd, namespace):
+    client = servicefabric_managed_client_factory(cmd.cli_ctx)
+    service = _safe_get_resource(client.services.get,
+                                 (namespace.resource_group_name, namespace.cluster_name,
+                                  namespace.application_name, namespace.service_name))
+
+    if service is None:
+        raise CLIError("Service '{}' Not Found.".format(namespace.service_name))
+    if service.properties.service_kind.lower() == ServiceKind.STATELESS.lower():
+        if namespace.metric_name is None or namespace.weight is None or namespace.default_load is None:
+            raise CLIError("--metric-name, --weight and --default-load are required")
+        if namespace.primary_default_load is not None or namespace.secondary_default_load is not None:
+            raise CLIError(
+                "--primary-default-load and --secondary-default-load can only be used for stateful services."
+            )
+        namespace.default_load = int(namespace.default_load)
+    elif service.properties.service_kind.lower() == ServiceKind.STATEFUL.lower():
+        if namespace.metric_name is None or namespace.weight is None or \
+           namespace.primary_default_load is None or namespace.secondary_default_load is None:
+            raise CLIError("--metric-name, --weight, --primary-default-load and --secondary-default-load are required")
+        if namespace.default_load is not None:
+            raise CLIError("--default-load can only be used for stateless services.")
```

Comment:
I recommend to use `ValidationError` instead of `CLIError`. Definition is in src/azure-cli-core/azure/cli/core/azclierror.py.

Reply:
Ok, I changed those